src/Core/Grid/Definition/Factory/AbstractIdentifiableGridDefinitionFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+/**
+ * Class AbstractIdentifiableGridDefinitionFactory implements identified grid definition creation.
+ */
+abstract class AbstractIdentifiableGridDefinitionFactory extends AbstractGridDefinitionFactory implements IdentifiableGridDefinitionFactoryInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * Grid definition already has an internal id used or hooks, that can also be used as an identifier for filters
+     */
+    public function getGridId(): string
+    {
+        return $this->getId();
+    }
```

Comment:
Still thinking it's confusing, we are already in a Grid context, the getId should be the getGridId. Should be renamed getFilterId